templates/assets.html

```diff
@@ -0,0 +1,77 @@
+<div class="assets">
+    <input type="checkbox" style="display: none" id="show-book-nav" />
+    <nav class="book-nav" role="navigation">
+        <ul class="book-nav-sections-container book-nav-sections">
+            {% for subsection in section.subsections %}
+            {% set section = get_section(path=subsection) %}
+            <li class="book-nav-section">
+                <a href="#{{ section.title | slugify }}" class="book-nav-section-title">{{ section.title }}</a>
+                {% if section.subsections %}
+                <ul class="book-nav-sections book-nav-sections-indented">
+                    {% set subsections = section.subsections %}
+                    {% if section.extra.sort_order_reversed %}
+                    {% set subsections = section.subsections | reverse %}
+                    {% endif %}
+                    {% for subsection in subsections %}
+                    {% set section = get_section(path=subsection) %}
+                    <li class="book-nav-section">
+                        <a href="#{{ section.title | slugify }}" class="book-nav-section-title">{{ section.title }}</a>
+                    </li>
+
+                    {% endfor %}
+                </ul>
+                {% endif %}
+            </li>
+            {% endfor %}
+        </ul>
+    </nav>
+    <div class="book-content">
+        {% for subsection in section.subsections %}
+        {% set section = get_section(path=subsection) %}
+
+        <h1 id="{{ section.title | slugify }}" class="asset-section">
+            {{ section.title }}<a class="anchor-link" href="#{{ section.title | slugify }}">#</a>
+        </h1>
+
+        {% set subsections = section.subsections %}
+        {% if section.extra.sort_order_reversed %}
+        {% set subsections = section.subsections | reverse %}
+        {% endif %}
+        {% for subsection in subsections %}
+        {% set section = get_section(path=subsection) %}
+
+        <h3 id="{{ section.title | slugify }}">
+            {{ section.title }}<a class="anchor-link" href="#{{ section.title | slugify }}">#</a>
+        </h3>
+        <div class="card-list padded-content">
+            {% set pages = section.pages %}
+            {% if section.extra.sort_order_reversed %}
+            {% set pages = section.pages | reverse %}
+            {% endif %}
+            {% for post in pages %}
+            <div class="card" id="{{ post.title | slugify }}">
+                <a href="{{ post.extra.link }}">
+                    {% if post.extra.image %}
+                    <div class="card-image card-image-default">
+                        <img src="{{ get_url(path=post.extra.image) }}" />
+                    </div>
+                    {% endif %}
+                    <div class="card-text">
+                        <div class="card-title">{{ post.title }}</div>
+                        <div class="card-description">
+                            {{ post.description | markdown(inline=true) | safe }}
```

Comment:
I'm not sure allowing arbitrary markdown here is a good idea. I think it gives users too much power over formatting / visibility (ex: we shouldn't let them slap a `# LOOK AT ME` h1 in their description). As it stands, we've only removed links "by convention". Theres nothing stopping users from adding links to their descriptions, other than the pull request process.

Reply:
I would like to keep markdown open, for at least code block, bold and italic, but that looks somewhat painful to validate automatically...

There's a manual review for PR now and the volume is ok, but I see how that won't work at scale and it wouldn't be nice to change the rule somewhere in the future to remove that.


I'l guess I'll remove it until I can find a way to validate that I like

Reply:
Things like codeblocks / bold / italic dont feel like much of a loss in the context of "short and sweet tagline describing a project". People shouldn't be including code examples in something like that. If we enable bold in a "short tagline" context, it incentivizes misuse to increase visibility (ex: just make the whole tagline bold to draw the user's attention away from non-bolded descriptions). 

I'd rather just keep this plain text, then enable markdown in the "long description" we show when someone clicks on an entry.

Reply:
I also think we should heavily limit character count for the "short description" / "tagline" / whatever we decide to call it. Stuff like this is too noisy / takes up too much space:

![image](https://user-images.githubusercontent.com/2694663/119902399-016eb280-befc-11eb-8c40-30381bd1ce80.png)

Reply:
ok 👍 

For long description: how do you feel about using `line-clamp` (https://caniuse.com/css-line-clamp)? It's pretty much the ellipsis text overflow for multiline text
<img width="294" alt="Screenshot 2021-05-28 at 00 45 34" src="https://user-images.githubusercontent.com/8672791/119906226-17a55e80-bf4e-11eb-81ce-35008de10b9e.png">

But it's not yet in the official css spec so it's supported by most browser with the `-webkit-` prefix. Most being all but IE

Reply:
Alternatively we could do it in Tera. We currently do this for the news section:

```rust
<div class="card-description">
  {{ page.summary | striptags | truncate(length=180) | safe}}
</div>
```

I think our "short description" should also have a max character limit enforced by the "bevy-asset repo toml validator". But theres no harm in truncating in the card, just to be doubly sure.

Reply:
(truncate does add ellipsis) https://tera.netlify.app/docs/#truncate

Reply:
Oh didn't even think to use the template engine for that 🤦 
But the limit in char count is not a guarantee that it wi display in a given multiline box due to line breaks

> I think our "short description" should also have a max character limit enforced by the "bevy-asset repo toml validator".

Then what happens fro existing assets that won't meet the new rules? we edit them ourselves? we remove them and ask for contributors to edit them?

Reply:
>  But the limit in char count is not a guarantee that it will display in a given multiline box due to line breaks

Rather than enforcing hard limits using line-clamp, I think we should probably just strictly define the size of the container div (for card size uniformity), disable overflow on that div, and then set character limit to something that prevents overflow in the first place. That ensures that worst case scenario, we allowed too many characters and they get cut off. Then we just need to either adjust the bounds to make those characters visible, or shrink the character limit.

Short term, I'm cool with just truncating in Zola and calling it a day. We can enforce card size uniformity when we rework the card layout.

> Then what happens fro existing assets that won't meet the new rules? we edit them ourselves? we remove them and ask for contributors to edit them?

We should edit them ourselves (and maybe leave a comment on the original pr to give the author a heads up in case they want to change something). Happy to do that work if nobody else wants to.

Reply:
Cutting at 100 chars seems to provide the best results 👍 

I added the task to update existing assets to the tracking issue